File: design-patterns/go/decorator/ui_element/border_decorator.go
// border_decorator.go

package main

import "fmt"

type BorderDecorator struct {
	*UIDecorator
}

func NewBorderDecorator(uiElement UIElement) (borderDecorator *BorderDecorator) {
	borderDecorator = &BorderDecorator{}
	borderDecorator.UIDecorator = NewUIDecorator(uiElement)
	return
}

func (borderDecorator *BorderDecorator) Draw() {
	// Can perform any additional task anywhere in the method

	borderDecorator.UIDecorator.Draw()

	// Write code to add border to the element
	fmt.Println("Adding Border to the element")
}

File: design-patterns/php/decorator/uielement/BorderDecorator.php
<?php
// BorderDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\UiElement;

class BorderDecorator extends UIDecorator {
    public function __construct(UIElement $uiElement) {
        parent::__construct($uiElement);
    }

    public function draw(): void {
        parent::draw();

        echo "Adding Border to the element\n";
    }
}